bindings/nodejs/lib/gpio.js

```diff
@@ -0,0 +1,112 @@
+/*
```

Comment:
ok, I remember that. But AFAIR when we talked during our F2F in Finland the "pure npm install" method wasn't to be addressed now. That is, evoking `npm install soletta` wouldn't be supported. The usage of "npm" was to be used inside our build to ease the process of creating the bindings (guest build system, not host build system).

In order to refresh our discussion, my arguments is that Soletta build is fairly complex and unless the user have all the required dependencies for all modules, then things would be silently disabled and resulting build would be very different from one to another. Like if the user happens to have "gtk-dev", then the "gtk" flow module will be there, otherwise it's disabled. And the list is long, see `data/json/dependencies.json`

In this approach, you should install to the system, as the build system is going away. We, the soletta developers, are the only folks that will ever see the build environment. I agree this is not the "usual workflow" for nodejs people, but given the above complexities it will never be the usual workflow in any way. Then for the specific case of soletta-node, the expected workflow for our users is to get this pre-installed via Ostro or ClearLinux. At this point if user did a package.json for npm that requires "soletta", it will just continue since the soletta is already there :-)

Then we agreed people would build Soletta and from it get all bindings, like Python, NodeJS. Hopefully we'll update our https://github.com/solettaproject/soletta-packaging to produce split packages with them, so if you're a Fedora user you can `dnf install soletta-nodejs` and it would pull in all the dependencies, placing them in the proper location such as `/usr/lib/node_modules` and so on.

Once things are more complete we can revisit the pure `npm install soletta` case, then we must also cover a way to force all dependencies to be available. Eventually that can happen with a different repository with this "host npm" that will fetch/include the actual soletta source with nodejs bindings, then it can have all the freedom to reshuffle things.

Makes sense?